src/java.desktop/share/classes/javax/swing/border/TitledBorder.java

```diff
@@ -350,10 +351,10 @@ public void paintBorder(Component c, Graphics g, int x, int y, int width, int he
                     if (g2 instanceof Graphics2D) {
                         Graphics2D g2d = (Graphics2D) g2;
                         Path2D path = new Path2D.Float();
-                        path.append(new Rectangle(borderX, borderY, borderW, labelY - borderY), false);
-                        path.append(new Rectangle(borderX, labelY, labelX - borderX - TEXT_SPACING, labelH), false);
-                        path.append(new Rectangle(labelX + labelW + TEXT_SPACING, labelY, borderX - labelX + borderW - labelW - TEXT_SPACING, labelH), false);
-                        path.append(new Rectangle(borderX, labelY + labelH, borderW, borderY - labelY + borderH - labelH), false);
+                        path.append(new Rectangle2D.Float((float) borderX, borderY, borderW, labelY - borderY), false);
```

Comment:
Why the current coordinates do not work, is it because we calculate them in the wrong way or probably the insets are wrong?

Reply:
I think there's a rounding error when doing x.5 scalings causing the drawing area to be one pixel off

Reply:
Can you please check how it will work for the undecorated frame, if it will be fine then the problem is somehow related to the size of the insets.

Reply:
The undecorated frame has no problems

Reply:
@mrserb I've been trying to manually adjust the insets/variables to see what could be causing the issue, but only decreasing borderX by 1 seems to fix the issue. Could you clarify what you can gather from the undecorated frame working?